app/classifier/drawing-tools/generic-tests.spec.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { expect } from 'chai';
+import drawingTools from '../drawing-tools';
+
+const x = 200;
+const y = 400;
+const cursors = [
+  {
+    x: 300,
+    y: 400
+  },
+  {
+    x: 250,
+    y: 450
+  },
+  {
+    x: 200,
+    y: 500
+  }
+];
+
+const props = {
+  color: 'red',
+  containerRect: {
+    height: 100,
+    width: 100
+  },
+  naturalHeight: 100,
+  naturalWidth: 100,
+  preferences: {
+    preferences: {}
+  },
+  scale: {
+    horizontal: 1,
+    vertical: 1
+  },
+  task: {
+    tools: [{ offsetX: 50, offsetY: 50 }]
+  }
+};
+
+for (const property in drawingTools) {
+  let wrapper;
+  const TaskComponent = drawingTools[property];
+
+  describe(`Task  ${property}`, function() {
+    let mark;
+
+    before(function () {
+      mark = TaskComponent.defaultValues({ x, y }, props);
+      mark.tool = 0;
+    });
+
+    it('should make changes with initStart', function() {
+      const initialMark = Object.assign({}, mark);
+      if (TaskComponent.initStart) {
+        TaskComponent.initStart(cursors[0], mark);
+        expect(initialMark).to.not.equal(mark);
```

Comment:
I think these static methods should be tested for return type too. Something along the lines of
```suggestion
        expect(TaskComponent.initStart(cursors[0], mark)).to.be.an('object');
```

Reply:
No idea why github suggested deleting the original line there. This was supposed to be an additional expectation.

Reply:
This will work for `initStart` and `initRelease`; however, `initMove` simply pushes some coordinates to a mark's points array without returning anything in a couple instances. The tests could be strengthened a bit once those components are rewritten to return the actual mark along with the push to `points`.

Reply:
The drawing task expects `initMove` (and `initStart` and `initRelease`) to return an object, which is merged into the mark: https://github.com/zooniverse/Panoptes-Front-End/blob/5473a7acacd4354dc7b550ba8e2d86d65b916c55/app/classifier/tasks/drawing/marking-initializer.cjsx#L78-L80

Problem is, if you're writing a drawing tool, you wouldn't know that without reading the drawing task code. 😞